File: ui-selenium/src/main/java/org/aeautomation/utils/TestDataGenerator.java
package org.aeautomation.utils;

import java.util.UUID;

public final class TestDataGenerator {

    private TestDataGenerator() {
    }

    public static String generateEmail(String prefix) {
        String domain = ConfigReader.getProperty("test.email.domain");
        return prefix + "_" + UUID.randomUUID().toString().replace("-", "") + "@" + domain;
    }
}

File: ui-playwright/utils/TestDataGenerator.ts
import { randomBytes } from 'crypto';
import ConfigReader from './ConfigReader';

/**
 * Utility class for generating dynamic test data at runtime.
 * Used for values that must be unique per test run (e.g. email addresses).
 */
class TestDataGenerator {

    /**
     * Generates a unique email address using a random hex string.
     * Format: {prefix}_{hex}@{domain}
     *
     * @param prefix - Short label identifying the test context (e.g. 'qa')
     * @returns Unique email address string
     */
    static generateEmail(prefix: string): string {
        const domain = ConfigReader.getEmailDomain();
        const unique = randomBytes(16).toString('hex');
        return `${prefix}_${unique}@${domain}`;
    }
}

export default TestDataGenerator;